app/src/main/java/com/alphawallet/app/widget/BalanceDisplayWidget.java

```diff
@@ -0,0 +1,77 @@
+package com.alphawallet.app.widget;
+
+import android.app.Activity;
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+
+import com.alphawallet.app.R;
+import com.alphawallet.app.entity.tokens.Token;
+import com.alphawallet.app.service.TokensService;
+import com.alphawallet.app.util.BalanceUtils;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * Created by Jenny Jingjing Li on 4/3/2021
+ */
+
+public class BalanceDisplayWidget extends LinearLayout
+{
+    public final TextView balance;
+    public final TextView newBalance;
+    private final ChainName chainName;
+    private final TokenIcon chainIcon;
+
+    private Token token;
+    private Activity activity;
+
+    public BalanceDisplayWidget(Context context, @Nullable AttributeSet attrs)
+    {
+        super(context, attrs);
+        inflate(context, R.layout.item_balance_display,this);
+        balance = findViewById(R.id.text_balance);
+        newBalance = findViewById(R.id.text_new_balance);
+        chainName = findViewById(R.id.chain_name);
+
+        token = null;
+
+        chainIcon = findViewById(R.id.chain_icon);
+
+    }
+
+    public void setupBalance(Token t, TokensService tokenService, Activity act)
+    {
+        token = t;
```

Comment:
token and activity don't need to be members, and you don't need Activity as the widget has Context to retrieve the resource.

Reply:
Actually Token does need to be a member, I didn't see the usage in the final function.